lib/note-content-editor.tsx

```diff
@@ -167,6 +173,86 @@ class NoteContentEditor extends Component<Props> {
     }
   };
 
+  completionProvider: (
+    selectedNoteId: T.EntityId | null
+  ) => languages.CompletionItemProvider = (selectedNoteId) => {
+    return {
+      triggerCharacters: ['['],
```

Comment:
it doesn't look like we're using this - is that just lost because the diff is hiding lines that aren't changing?